### PR TITLE
LPS-71409

### PR DESCRIPTION
**Pull request by ericyanLr**
Hi @SamZiemer,

Original Tickets:
- [LPP-24542 Login information over GET request can result in unauthorized access](https://issues.liferay.com/browse/LPP-24542)
- [LPS-71409 Users can login using GET method with Login portlet and /c/portal/login](https://issues.liferay.com/browse/LPS-71409)

Original Pull Request: 
- https://github.com/SamZiemer/liferay-portal/pull/220

### Summary
This pull request is a re-send and takes into consideration of the previous reviewer's comment:
>Hi guys,
thanks for the pull but this is not the fix we need.
\
Anybody can send POST /c/portal/login?login=test@liferay.com&password=test&redirect=%2F HTTP/1.1
\
What we want to prevent is having password in URL. Please resend.
Thank you.

Reference:https://github.com/topolik/liferay-portal/pull/429#issuecomment-291471319

### New Approach
Our previous approach prevented the **password parameter** from being passed from the URL for a `GET` request.

Our interpretation of the reviewer's comment is that we should also prevent the **password parameter** from being passed from the URL for a `POST` request as well.

The new approach used in this PR basically rejects the login request if a **password parameter** is being passed from the URL, regardless of what request method is used.
* Basically, we will not attempt to login if we detect the **password parameter** in the URL.

----
Please let me know if you have any questions.
Thanks!

### CODE REVIEW

**Comment by SamZiemer** [comment]
Forwarded to https://github.com/jonathanmccann/liferay-portal/pull/1287
